Hobby Page, add content and slideshow.

jessdejong opened the pull request:
Create hobbies.html, add a slideshow and implement it in javascript, and style the slideshow.

daniel-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,81 @@ function addRandomGreeting() {
+    ['basketball.JPG', 'chess.JPG', 'crochet.JPG', 'smoothie.jpg', 'statefarm.JPG',
+    'tennis.JPG', 'UIL_state.JPG', "Uke.jpg"];
+
+const descriptions =
```
instead of maintaining parallel arrays, group related data together in object form, e.g.

```js
const images = [
  {name: 'basketball.jpg', label: 'Playing basketball with the Tennis Intramural Team!'},
  // ...
];
```

Rationale is that it's easier to keep a single list in-sync, and you can pass around related data together instead of dealing with indices.

jessdejong replied:
Makes sense, I'll make the change!

daniel-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,81 @@ function addRandomGreeting() {
+/* Onload, initalize slide show with an image */
+function initializeSlideShow() {
+  // display the first image in the slideshow
+  let firstImageURL = 'images/basketball.JPG';
```
here's an example of where the above suggestion is useful :)

```js
let firstImage = images[0];
displayNewSlide(firstImage);
```

daniel-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,81 @@ function addRandomGreeting() {
+    slideShowIndex = (slideShowIndex + 1) % images.length;
+  }
+  else if (direction == 'down') {
+    slideShowIndex = slideShowIndex - 1;
```
nit: make 'up', 'down', 'random', etc. into constants, as it will be easier to find bugs

```js
const DIRECTION_UP = 'up';
// ...
updateSlideShowIndex(DIRECTION_UP);
```

(typos will be surfaced as unrecognized name errors instead of being silently ignored)

jessdejong replied:
Fixed!

daniel-google commented on portfolio/src/main/webapp/style.css:
```diff
@@ -58,3 +58,41 @@ body {
+}
+
+.slideshowNavigation button {
+  background-color: dodgerblue;
```
is this an actual CSS color??

jessdejong replied:
LOL yes, it is!

daniel-google commented on portfolio/src/main/webapp/style.css:
```diff
@@ -58,3 +58,41 @@ body {
+  width: 100px;
+}
+
+#slideshow-content-container img {
```
nit: use consistent naming convention for identifiers, either all `camelCase` or `dashed-words`

jessdejong replied:
Fixed

jessdejong commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,81 @@ function addRandomGreeting() {
+
+const images =
+    ['basketball.JPG', 'chess.JPG', 'crochet.JPG', 'smoothie.jpg', 'statefarm.JPG',
+    'tennis.JPG', 'UIL_state.JPG', "Uke.jpg"];
```
Fixed!

jessdejong commented:
Standardized the image extension to .jpg,
added direction constants,
standardized code to camelCase,
and replaced the image and description lists with a single list, images

Let me know if there is anything else!

daniel-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -23,10 +23,80 @@ function addRandomGreeting() {
+  }
+  else if (direction == RANDOM_SLIDE) {
+    slideShowIndex = Math.floor(Math.random() * images.length);
+  }
```
nit: consider adding an `else{}` block that throws an exception or logs an error to the console

jessdejong replied:
Good idea! I'll go ahead and add it.

daniel-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -23,10 +23,80 @@ function addRandomGreeting() {
+
+/* Navigating the Slideshow with an index */
+function updateSlideShowIndex(direction) {
+  if (direction == DIRECTION_UP) {
+    slideShowIndex = (slideShowIndex + 1) % images.length;
+  }
+  else if (direction == DIRECTION_DOWN) {
+    slideShowIndex = slideShowIndex - 1;
+    if (slideShowIndex < 0) {
+      slideShowIndex += images.length;
+    }
+  }
+  else if (direction == RANDOM_SLIDE) {
+    slideShowIndex = Math.floor(Math.random() * images.length);
+  }
```
nit: consider making this whole thing a `switch(){}` statement

jessdejong replied:
Made the change! The code looked more organized. Are switch statements a little bit faster?

jessdejong commented:
My additional commits addressed your suggestions:
Throw an exception to handle invalid parameters,
and replace an if/else if/else with a switch statement

jessdejong commented:
Since both of my additional commits just resolved the two code suggestions you gave me, I'm going to go ahead and merge this. Let me know if I shouldn't have, or if you have any additional comments! Thanks.
